lighthouse-core/runner.js

```diff
@@ -259,10 +259,19 @@ class Runner {
         output: undefined,
         channel: undefined,
         budgets: undefined,
+        skipAudits: undefined,
+        onlyAudits: undefined,
+        onlyCategories: undefined,
+        extraHeaders: undefined,
       };
       const normalizedGatherSettings = Object.assign({}, artifacts.settings, overrides);
       const normalizedAuditSettings = Object.assign({}, settings, overrides);
 
+      for (const k of Object.keys(normalizedGatherSettings)) {
+        if (!isDeepEqual(normalizedGatherSettings[k], normalizedAuditSettings[k])) {
```

Comment:
Why doesn't `isDeepEqual` cover this in the existing code? Is it just to get a more specific error message?

Reply:
Yea, I could remove it but it helped a bit with debugging sooo

Reply:
Could you add a comment about why we need two, like "First one checks equality of keys, second one checks for missing keys".